Assets/Scripts/Weapons/WeaponPickup.cs

```diff
@@ -0,0 +1,49 @@
+using UnityEngine;
+using System.Collections.Generic;
+using System.Collections;
+
+
+public class WeaponPickup : MonoBehaviour
+{
+    public WeaponData weaponData;  // Assign the weapon data in the inspector
+
+    private bool isColliding;
+    
+    void OnTriggerEnter(Collider other)
+    {        
+        // if(isColliding) return;
+        //     isColliding = true;
+
+        if (other.CompareTag("Player"))  // Ensure the player has a "Player" tag
```

Comment:
It might be easier to use the Static Tag Strings so that if the player tag ever changes we will only need to change it in one place.